app/deployments/routes.py

```diff
@@ -574,14 +601,18 @@ def delnode(depid):
     if dep is not None and dep.physicalId is not None:
         try:
             vm_id = request.args["vmid"]
-            app.logger.debug(f"Requested deletion of node {vm_id} on deployment {dep.uuid}")
+            app.logger.debug(
+                f"Requested deletion of node {vm_id} on deployment {dep.uuid}"
+            )
             resource = app.orchestrator.get_resource(access_token, depid, vm_id)
             resources = app.orchestrator.get_resources(access_token, depid)
 
             node = next((res for res in resources if res.get("uuid") == vm_id), None)
             node_name = node.get("toscaNodeName")
             # current count -1 --> remove one node
-            count = sum(1 for res in resources if res.get("toscaNodeName") == node_name) - 1
+            count = (
+                sum(1 for res in resources if res.get("toscaNodeName") == node_name) - 1
```

Comment:
This `sum` of +1 is pretty bizzare :)
Since we are creating a new list anyway of "ones", instead we can filter the node list and simply get the length

```python
filtered = list(filter(lambda res: res.get("toscaNodeName") == node_name)
count = len(filtered) - 1
```
Are we sure that that "-1" is correct? When we have resources matching the `node_name` that counter has value -1, which is bizarre.